tests/XTMF2.UnitTests/TestRun.cs

```diff
@@ -216,32 +214,225 @@ public void ReportRunProgress()
+                    runBus.ClientErrorWhenRunningModelSystem += (sender, runId, e, stack) =>
+                    {
+                        error = e + "\r\n" + stack;
+                        sim.Release();
+                    };
+                    var runDirectory = Path.Combine(pSession.RunsDirectory, "RunResultsCompleted");
+                    Assert.IsTrue(runBus.RunModelSystem(msSession, runDirectory, "Start", out var id, ref error), error);
+                    // give the models system some time to complete
+                    if (!sim.Wait(2000))
+                    {
+                        Assert.Fail("The model system failed to execute in time!");
                     }
+                    Assert.IsTrue(success, "The model system failed to execute to success! " + error);
+
+                    // Check to make sure that the meta-data was stored in the run directory
+                    var results = new RunResults(runDirectory);
+                    Assert.IsTrue(results.Completed, "The run results stated that the run failed to complete!");
+                    Assert.IsFalse(results.HasError, "The model system that successfully ran has an error!");
+                    Assert.IsNull(results.Error, "There was an error for a successfully run model system!");
+                    Assert.IsNull(results.ErrorMessage, "There was an error message for a successfully run model system!");
+                    Assert.IsNull(results.ErrorStackTrace, "There was an error stack trace for a successfully run model system!");
+                    Assert.IsNull(results.ErrorModuleName, "There was a module's name referred to as a source of an error in a successfully run model system.");
+                });
+            });
+        }
+
+        [TestMethod]
+        public void RunResultsRuntimeError()
+        {
+            RunInModelSystemContext("RunResultsRuntimeError", (user, pSession, msSession) =>
+            {
+                string error2 = null;
+                var ms = msSession.ModelSystem;
+                Assert.IsTrue(msSession.AddModelSystemStart(user, ms.GlobalBoundary, "Start", out Start start, ref error2), error2);
+                Assert.IsTrue(msSession.AddNode(user, ms.GlobalBoundary, "RequiresAChild",
+                typeof(FailsAtRuntime), out var node, ref error2), error2);
+                Assert.IsTrue(msSession.AddLink(user, start, start.Hooks[0], node, out var link, ref error2), error2);
+                CreateRunClient(true, (runBus) =>
+                {
+                    string error = null;
+                    bool success = false;
+                    using SemaphoreSlim sim = new SemaphoreSlim(0);
+
+                    runBus.ClientFinishedModelSystem += (sender, e) =>
+                    {
+                        success = true;
+                        sim.Release();
+                    };
+                    runBus.ClientErrorWhenRunningModelSystem += (sender, runId, e, stack) =>
```

Comment:
Does it make sense to use OnErrorXYZ language to show it's an event?

Reply:
Following Microsoft style for WPF it looks like events do not get the OnXYZ pattern. On is included for methods for those events that can be overridden.